src/API.jl

```diff
@@ -1093,6 +1092,14 @@ function precompile(ctx::Context; internal_call::Bool=false)
     finished = true
     notify(first_started) # in case of no-op
     wait(t_print)
+    failed_direct = filter(in(direct_deps), failed_deps)
+    if !isempty(failed_direct)
+        failed_list = ""
+        for d in failed_direct
+            failed_list *= "  $d\n"
+        end
+        throw(ErrorException("The following direct dependencies failed to precompile:\n$(failed_list)"))
```

Comment:
```suggestion
        pkgerror("The following direct dependencies failed to precompile:\n$(failed_list)"))
```

(the difference is that it doesn't show a backtrace when you call it from the Pkg REPL)

Reply:
perfect, thanks